src/main/java/com/aws/greengrass/iot/IotCloudHelper.java

```diff
@@ -46,9 +47,11 @@
      * @param body        Http body for the request
      * @return Http response corresponding to http request for path
      * @throws AWSIotException when unable to send the request successfully
+     * @throws DeviceConfigurationException when device is not configured with credential endpoint
      */
     public IotCloudResponse sendHttpRequest(final IotConnectionManager connManager, String thingName, final String path,
-                                            final String verb, final byte[] body) throws AWSIotException {
+                                            final String verb, final byte[] body)
+                                            throws AWSIotException, DeviceConfigurationException {
```

Comment:
this is not compatible. Wrap the exception instead.

Reply:
Compatible with ? Can you describe more?

Reply:
any plugin which uses it. This change is not binary compatible